### ISSUE TITLE
Two models with cross loss function

### ISSUE THREAD

**Issue by cpivot** [keras-team/keras#11416]
Hi, 
I'm working on a heteroscedactic neural network. 
What I want is to get two neural networks and the loss function of each NN depends on the another NN. 
Here a short example : 

```
   def define_NN():
        init=keras.initializers.RandomNormal(mean=0.0, stddev=1., seed=None)
        input_tensor=(Input(shape=(X.shape[1],)))
        denseA=Dense(15,activation='tanh')(input_tensor)
        denseB=Dense(7,activation='tanh')(denseA)
        output_tensor=Dense(1,activation='linear')(denseB)
        return Model(inputs=input_tensor,outputs=output_tensor)


    def customLossSigma(modelMu,x):
        def neg_log_likelihood_Sigma(y_true, y_pred):
            mu=modelMu(x)
            inter=(mu-y_true)/K.clip(y_pred,K.epsilon(),None)
            val=K.log(K.clip(K.square(y_pred),K.epsilon(),None))+K.square(inter)
            return val
        return neg_log_likelihood_Sigma

    def customLossMu(modelSigma,x):
        def neg_log_likelihood_Mu(y_true, y_pred):
            sigma=modelSigma(x)
            inter=(y_pred-y_true)/K.clip(sigma,K.epsilon(),None)
            val=K.log(K.clip(K.square(sigma),K.epsilon(),None))+K.square(inter)
            return val
        return neg_log_likelihood_Mu

    Adam=keras.optimizers.Adam(lr=learning_rate)

    modelMu=define_NN()
    modelSigma=define_NN()

    modelSigma.compile(loss=customLossSigma(modelMu,modelSigma.input), optimizer=Adam)
    modelMu.compile(loss=customLossMu(modelSigma,modelMu.input), optimizer=Adam)

    for ii in range(3):
        modelMu.fit(X_train, y_train, epochs=nepochs, batch_size=15,verbose=verbose,validation_data=(X_test, y_test))
        modelSigma.fit(X_train, y_train, epochs=nepochs, batch_size=15,verbose=verbose,validation_data=(X_test, y_test))
```

Is there a way to do so? Either by define in a correct way my loss functions or by define in another way my NN? 

Thank's for your help
Charles
